openapi_tester/validators.py

```diff
@@ -74,24 +73,64 @@ def wrapped(value: Any):
 def validate_type(schema_section: Dict[str, Any], data: Any) -> Optional[str]:
     schema_type: str = schema_section.get("type", "object")
     if not VALIDATOR_MAP[schema_type](data):
+        an_articles = ["integer", "object", "array"]
         return VALIDATE_TYPE_ERROR.format(
-            expected=OPENAPI_PYTHON_MAPPING[schema_type],
-            received=type(data).__name__,
+            article="a" if schema_type not in an_articles else "an",
+            type=schema_type,
+            received=f'"{data}"' if isinstance(data, str) else data,
         )
     return None
 
 
 def validate_format(schema_section: Dict[str, Any], data: Any) -> Optional[str]:
     schema_format: str = schema_section.get("format", "")
     if schema_format in VALIDATOR_MAP and not VALIDATOR_MAP[schema_format](data):
-        return VALIDATE_FORMAT_ERROR.format(expected=schema_section["format"], received=str(data))
+        example_values = {
+            "byte": {"value": b"example", "article": "a"},
+            "base64": {"value": b"ZXhhbXBsZQ==", "article": "a"},
+            "date": {"value": '"2020-01-22"', "article": "a"},
+            "date-time": {"value": '"2020-01-22 08:00"', "article": "a"},
+            "double": {"value": "2.22", "article": "a"},
+            "email": {"value": '"example@gmail.com"', "article": "an"},
+            "float": {"value": "2.2", "article": "a"},
+            "ipv4": {"value": '"192.0.2.235"', "article": "an"},
+            "ipv6": {"value": '"2001:0db8:85a3:0000:0000:8a2e:0370:7334"', "article": "an"},
+            "time": {"value": '"20:00:00"', "article": "a"},
+            "uri": {"value": '"https://example.com/"', "article": "a"},
+            "url": {"value": '"https://example.com/"', "article": "a"},
+        }
+        return VALIDATE_FORMAT_ERROR.format(
+            format=schema_format,
+            article=example_values[schema_format]["article"],  # type: ignore
+            example=example_values[schema_format]["value"],  # type: ignore
+            received=f'"{data}"',
+        )
     return None
 
 
+def pretty_print_list(_list: list) -> str:
```

Comment:
please remove this method completely. 

You can just use the excellent built-in library called pprint: https://docs.python.org/3/library/pprint.html

Reply:
Yep removing it as we changed the enum string anyways.